src/database/common.c

```diff
@@ -246,17 +246,6 @@ void db_init(void)
 	// explicitly check for failures to have happened
 	sqlite3_config(SQLITE_CONFIG_LOG, SQLite3LogCallback, NULL);
 
-	// SQLITE_DBCONFIG_DEFENSIVE
-	// Disbale language features that allow ordinary SQL to deliberately corrupt
-	// the database file. The disabled features include but are not limited to
-	// the following:
-	//
-	// - The PRAGMA writable_schema=ON statement.
-	// - The PRAGMA journal_mode=OFF statement.
-	// - Writes to the sqlite_dbpage virtual table.
-	// - Direct writes to shadow tables.
-	sqlite3_config(SQLITE_DBCONFIG_DEFENSIVE, true);
```

Comment:
Why did you remove this? I see, that `PRAGMA writable_schema` is now set if needed and switched off afterwards. But what's about the other options set by the flag?

Reply:
It is not possible to set `writable_schema` at all in defensive mode. It is neither possible to disable defensive mode when it was activated once. As we are in full control of all SQL ran against the database and users cannot inject malicious code anywhere, this is superfluous and we should just remove it.